lib/middleware/runner.rb

```diff
@@ -42,17 +42,18 @@ def build_call_chain(stack)
       # is always the empty middleware, which does nothing but return.
       stack.reverse.inject(EMPTY_MIDDLEWARE) do |next_middleware, current_middleware|
         # Unpack the actual item
-        klass, args, block = current_middleware
+        klass, args, kwargs, block = current_middleware
 
         # Default the arguments to an empty array. Otherwise in Ruby 1.8
         # a `nil` args will actually pass `nil` into the class. Not what
         # we want!
         args ||= []
+        kwargs ||= {}
```

Comment:
I only added this line because runner_spec omits args, and only passes in `klass`. This wouldn't happen if builder is used.

However, if we no longer support Ruby 1.8, I wonder if we should remove this block all together, and amend runner_spec instead?